src/wallet/wallet.go

```diff
@@ -178,6 +178,77 @@ func NewWallet(wltName string, opts Options) (*Wallet, error) {
 	return w, nil
 }
 
+// NewWalletWithScan create wallet and scan ahead N address
+func NewWalletWithScan(wltName string, opts Options, scanN uint64, bg BalanceGetter) (*Wallet, error) {
```

Comment:
is this almost the same as `NewWallet`?  If so, can use something like `newWallet()` internal method with an option to scan